### ISSUE TITLE
Stats: Enhance post details page

### ISSUE THREAD

**Issue by jeffgolenski** [Automattic/wp-calypso#70671]
# Development

### Must have:
- [x] Restyle the "Post Likes" section into a card.
( https://github.com/Automattic/wp-calypso/pull/71305 )
- [x] Apply the chart component from the Traffic page onto the post details page.
( https://github.com/Automattic/wp-calypso/pull/71257 )
  - Only showing view counts. Do not add support for visitors, likes, or comments for the first iteration.
- [x] Apply the All-time Insight component from the Insights page on the post details page.
( https://github.com/Automattic/wp-calypso/pull/71164, https://github.com/Automattic/wp-calypso/pull/71465 )
  - [x] Ensure that the "Months and Years" and "Average per Day" buttons are present.
  - [x] Add support for the "Recent Weeks" button
- [x] Create an "All-time stats" card using the [post stats card component](https://github.com/Automattic/wp-calypso/issues/70852).
( https://github.com/Automattic/wp-calypso/pull/71305 )

--- Updated ---

- [x] Append the `Jetpack Colophon` along the bottom of this post details page.
( https://github.com/Automattic/wp-calypso/pull/72319 )

### Nice to have:
- ~[ ] Add support for Views, Visitors, Likes, and Comments tabs under the chart component.~
( https://github.com/Automattic/wp-calypso/issues/70671#issuecomment-1348024629 )

> > Maybe we could only polish the used chart component and the single chart tab views?

> That sounds good to me and we can even drop the nav tabs altogether + add the "Views" description to the section title like so:

Must-have items must be completed before closing this issue. Please spin off any unfinished tasks into separate issues.

# Design

The detail pages of each post are quite antiquated. Please break this down into as many PRs are you need and let myself or @keoshi know if you want us to break down the issues further!

These detail pages reuse a lot of the components for the Traffic and Insights pages, so we don't need to reinvent the wheel there. 

|Before|After|
|-|-|
|![Image](https://user-images.githubusercontent.com/214813/205173885-89fd29e4-a696-4613-9ec4-ea994b3fdff2.png)|![Image](https://user-images.githubusercontent.com/214813/205174008-3a573794-6621-46ba-8fdd-12826187b26c.png)|

**Comment by dognose24** [comment]
> Apply the chart component from the Traffic page onto the post details page.

> Add support for Views, Visitors, Likes, and Comments tabs under the chart component.

After surveying the post data of this detail page, the data to display the post `visitors`, `likes`, and `comments` for a certain duration seem not to exist for now. So I think we could not provide the whole chart with tabs components as the `Traffic` page in this modernization iteration. Maybe we could only polish the used chart component and the single chart tab `views`? cc @jsnmoon @jeffgolenski @keoshi

**Comment by jsnmoon** [comment]
@dognose24: The visitors, likes, and comments chart tabs are not a must-have feature for this task. Let's punt it into a follow-up GitHub issue if you believe it won't be straightforward to implement.

**Comment by keoshi** [comment]
> Maybe we could only polish the used chart component and the single chart tab `views`?

That sounds good to me and we can even drop the nav tabs altogether + add the "Views" description to the section title like so:

<img width="1694" alt="image" src="https://user-images.githubusercontent.com/390760/207278122-4b6560e9-ed0e-476e-87b3-b7d6f334f91e.png">

**Comment by dognose24** [comment]
> we can even drop the nav tabs altogether + add the "Views" description to the section title

That looks more concise for Views only. Thanks for the information!

**Comment by dognose24** [comment]
Regarding PR #71257, I was trying to add period navigation arrows. However, the data manipulation inside `StatsPostSummary` is not acceptable for the existing `StatsPeriodNavigation` handling. And the state refactor would be too complicated before removing the feature gate. Since clicking on the bar can change the selected date, I will tackle the period navigation arrows in another follow-up PR. cc @jsnmoon @jeffgolenski @keoshi

**Comment by jsnmoon** [comment]
@dognose24: Sounds good to me!

**Comment by jsnmoon** [comment]
@dognose24: I'd appreciate it if you could update this parent issue with our current progress 🙂

**Comment by jsnmoon** [comment]
@dognose24, I want to report three defects:
> <img width="1090" alt="image" src="https://user-images.githubusercontent.com/4044428/209253990-60e8dcca-11f7-4c38-9643-54819d75bf01.png">
- [x] All-time stats card of the Highlights section is broken for "Hompage / Archives".
( https://github.com/Automattic/wp-calypso/pull/73459 )
- [x] The empty "likes" card is missing the "There are no likes on this post yet" copy.
( https://github.com/Automattic/wp-calypso/pull/73459 )

> <img width="1083" alt="image" src="https://user-images.githubusercontent.com/4044428/209254091-6e0ec6c6-a155-44ff-a256-b0bfdcc64cc1.png">
- [x] The thumbnail image doesn't stretch fill the full height of the card.
( https://github.com/Automattic/wp-calypso/pull/71553 )

@keoshi & @jeffgolenski: The empty "likes" card doesn't provide much value. Any ideas on how to improve this? Maybe we can encourage them to blaze it? 😉

**Comment by dognose24** [comment]
Two issues were coming from [the PR response](https://github.com/Automattic/wp-calypso/pull/71305#pullrequestreview-1222421782):

- [x] 1. When the image is too high, the numbers/image flow out of the section:
( https://github.com/Automattic/wp-calypso/pull/71553 )

<img width="672" alt="image" src="https://user-images.githubusercontent.com/1425433/208373509-0ff357f8-85a9-45cc-902d-c9de76cbb76d.png">

- [x] 2. When the page width is less than about 570px, the right border of the highlight section disappears:
( https://github.com/Automattic/wp-calypso/pull/71530 )

<img width="474" alt="image" src="https://user-images.githubusercontent.com/1425433/208373826-e6a5366a-0b19-4bc4-b68d-4bf38aa76204.png">

**Comment by dognose24** [comment]
- [x] [The empty cell should be colored on the Recent weeks' table](https://github.com/Automattic/wp-calypso/pull/71465#issuecomment-1363413506):
( https://github.com/Automattic/wp-calypso/pull/71552 )

> I've attached a screenshot to illustrate something we could fix in a follow-up PR. It doesn't make sense to shade cells with no data.

> <img width="998" alt="image" src="https://user-images.githubusercontent.com/4044428/209237847-ba917d8f-bc7a-4de1-9ac4-460f0297815b.png">

**Comment by dognose24** [comment]
> Another wonderful addition. Thanks for this. The only issue I'm seeing right now is that the All-time stats box breaks the layout when the title is too long. https://github.com/Automattic/wp-calypso/pull/71530#issuecomment-1371420470

Example: This post title in english occupies 3 lines, which pushes the card to break the balance with the card next to it

![Screen Shot 2023-01-04 at 4 04 01 PM](https://user-images.githubusercontent.com/214813/210649932-46fe7650-6a24-40c5-8423-144b4c60d1af.png)

Can we:

- [x] Limit the displayed title to 48 characters max (not including spaces) and then insert an ellipsis "..." when it hits that max character count?
( https://github.com/Automattic/wp-calypso/pull/72316 )

So it looks like this...
![Screen Shot 2023-01-04 at 4 04 41 PM](https://user-images.githubusercontent.com/214813/210650200-bbef79e9-5979-4735-a5db-23fc7288b597.png)

### Mobile

I'm also noticing that we need a bit of work on the mobile views (tablet and phone). 

![Screen Shot 2023-01-04 at 4 12 11 PM](https://user-images.githubusercontent.com/214813/210651089-90eaec56-ead4-44e6-975c-09f0be796182.png)
![Screen Shot 2023-01-04 at 4 11 53 PM](https://user-images.githubusercontent.com/214813/210651091-519098d6-d0c1-40c9-815f-21fba7e7726f.png)


Can we:

- [x] Make both of these cards full width at a higher tablet breakpoint? 
( https://github.com/Automattic/wp-calypso/pull/72316 )
- [x] Ensure the featured image is displayed at 84 x 84 pixels on mobile?
( https://github.com/Automattic/wp-calypso/pull/72316 )

![Screen Shot 2023-01-04 at 4 15 07 PM](https://user-images.githubusercontent.com/214813/210651346-9c066295-e517-4f8c-97b5-7345bf04a9c3.png)

**Comment by dognose24** [comment]
All related issues are addressed.
